src/backend/stream.rs

```diff
@@ -341,6 +345,17 @@ impl<'ctx> PulseStream<'ctx> {
             if stm.input_stream.is_some() {
                 let nframes = nbytes / stm.output_sample_spec.frame_size();
                 let nsamples_input = nframes * stm.input_sample_spec.channels as usize;
+                let val = stm.output_frame_count.fetch_add(nframes, Ordering::SeqCst);
+                if val == 0 {
+                    let input_fc = stm.input_frame_count.load(Ordering::SeqCst);
+                    if input_fc > nframes {
+                        // The ringbuffer contains old data
+                        // Remove them to decrease initial latency
+                        let nframes_to_drop = input_fc - nframes;
+                        cubeb_logv!("Dropping {} frames in the buffer", nframes_to_drop);
+                        stm.input_buffer_manager.as_mut().unwrap().get_linear_input_data(nframes_to_drop);
+                    }
+                }
```

Comment:
https://github.com/ChunMinChang/cubeb-coreaudio-rs/blob/trailblazer/src/backend/mod.rs#L625-L632 is the same and is clearer. Overloading `get_linear_input_data` to trim the buffer works but is not explicit. Also you need to adjust the atomic I think.